pkg/ocm/machines/machines.go

```diff
@@ -53,12 +59,20 @@ func GetMachineTypes(client *cmv1.Client) (machineTypes []*cmv1.MachineType, err
 }
 
 // Validate AWS machine types
-func ValidateMachineType(machineType string, machineTypeList []string) (string, error) {
+func ValidateMachineType(machineType string, machineTypes []*MachineType, multiAZ bool) (string, error) {
 	if machineType != "" {
+		var machineTypeList []string
 		// Check and set the cluster machineType
 		hasMachineType := false
-		for _, v := range machineTypeList {
-			if v == machineType {
+		for _, v := range machineTypes {
+			machineTypeList = append(machineTypeList, v.MachineType.ID())
+			if v.MachineType.ID() == machineType {
+				if v.MachineType.Category() == AccelatedComputing {
+					if v.AvailableQuota < getDefaultNodes(multiAZ) {
```

Comment:
This can all be a single conditional:
```go
if v.MachineType.Category() == AcceleratedComputing && v.AvailableQuota < getDefaultNodes(multiAZ) {
  err := ...
}
```